packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,276 @@
+
+### `BottomSheetFooterLeading` API
+
+| Prop     | Type            | Default     | Description                          | Required |
+| -------- | --------------- | ----------- | ------------------------------------ | -------- |
+| `title`  | `string`        | `undefined` | Title of the BottomSheet footer      | ✅       |
+| `prefix` | `IconComponent` | `undefined` | Prefix icon placed before title text |          |
+
+### `BottomSheetFooterTrailing` API
+
+| Prop      | Type                                     | Default     | Description                                | Required |
+| --------- | ---------------------------------------- | ----------- | ------------------------------------------ | -------- |
+| `actions` | `{ primary: Action, secondary: Action }` | `undefined` | Renders a primary/secondary action buttons |          |
+
+```ts
+type Action = {
+  onClick: () => void;
+  text: string;
+  type?: 'button' | 'reset' | 'submit';
+  accessibilityLabel: string;
+  isLoading: boolean;
+  isDisabled: boolean;
+  icon: React.ReactNode;
+  iconPosition: 'left' | 'right';
+};
+```
+
+## Composition with Dropdown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
+
+```jsx
+import { Spinner, useTheme, useBreakpoint } from "@razorpay/blade";
+
+const BottomSheet = React.lazy();
+const DropdownOverlay = React.lazy();
+
+const App = () => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" />
+      {/* We can either put a fallback spinner or show skeleton loaders */}
+      <React.Suspense fallback={<Spinner />}>
+        {isMobile ? (
+          <BottomSheet>
+            <BottomSheetHeader>
+              <BottomSheetHeaderLeading title="Payments Links" prefix={PayIcon} />
```

Comment:
1. **Bundle size doubts**
In this API, we're using `<Dropdown />` so it will add the Dropdown's bundle right? 🤔 The only component I see we're code-splitting is `<DropdownOverlay />` which should be less than 2kb since it is just that modal that opens. It doesn't really have any major logic.\
\
All the critical Dropdown logic is in Dropdown component inside `useDropdown` hook and utilities.


2. **Should we have it controlled?**
  Dropdown is uncontrolled and open-close logic of Dropdown is handled by Dropdown itself. So the `open={}` API seems a bit different to me compared to Dropdown. It's like the component on Desktop handles state internally but the similar component of mobile handles it externally.\
\
  Apart from that, you might have to add `aria-expanded` or `aria-controls` to the trigger right? if we give the state management to consumer I would assume the consumer has to handle that?

I wanted to suggest something like this so that we take care of above issues- 
```jsx
<BottomSheet>
  <SelectInput label="Select Action" />
  <BottomSheetContainer>
     <BottomSheetHeader />
     <ActionList />
     <BottomSheetFooter />
  </BottomSheetContainer>
</BottomSheet>
```

Then we'll also be able to do
```jsx
const ResponsiveDropdown = isMobile ? BottomSheet : Dropdown;
const ResponsiveDropdownOverlay = isMobile ? BottomSheetContainer : DropdownOverlay;
const ResponsiveBottomSheetHeader = isMobile ? BottomSheetHeader : React.Fragament;
const ResponsiveBottomSheetFooter = isMobile ? BottomSheetFooter : React.Fragment;

// ...

<ResponsiveDropdown>
   <SelectInput />
   <ResponsiveDropdownOverlay>
      <ResponsiveBottomSheetHeader />
      <ActionList />
      <ResponsiveBottomSheetFooter />
    </ResponsiveDropdownOverlay>
</ResponsiveDropdown>



```

what do you think?